packages/daemon/src/server.ts

```diff
@@ -17,40 +17,59 @@ export type CreateServerOptions = {
    */
   frontendDistPath: string;
   /**
-   * Listen Express on port
+   * Listen Fastify on port
    */
   port: number;
 };
 
-export const createServer = ({ frontendDistPath, port }: CreateServerOptions) => {
-  const app: Express = express();
-  let server: ReturnType<typeof app.listen> | null = null;
+export const createServer = (params: CreateServerOptions) => {
+  const fastify = Fastify({
+    logger: true
+  });
+  const staticDir = path.join(__dirname, params.frontendDistPath);
+
+  try {
+    fastify.register(fastifyCors, {
+      origin: true,
+    });
+
+    fastify.register(fastifyStatic, {
+      root: staticDir,
+      prefix: '/', // optional: default '/'
+    });
+  } catch (err) {
+    fastify.log.error('Error registering plugins:', err);
+    process.exit(1);
+  }
```

Comment:
```suggestion
fastify.register(fastifyCors, {
  origin: true,
});

fastify.register(fastifyStatic, {
  root: staticDir,
  prefix: '/', // optional: default '/'
});
```

Let's keep it simple. No need for `try/catch` here. We can catch the error at a higher level.